tests/exercise_instructions.rs

```diff
@@ -0,0 +1,506 @@
+                        addr >> 32,
+                    ),
+                    &mut prng,
+                    25,
+                );
+                test_ins(
+                    format!(
+                        "mov32 r{tmp},{}
+                         mov64 r{src},{:#x}
+                         lsh64 r{src},32
+                         or64 r{src},r{tmp}
+                         stxdw [r{src}{offset:+}], r{dst}",
+                        addr as i32,
+                        addr >> 32,
+                    ),
+                    &mut prng,
+                    25,
+                );
+
+                // test conditionals
+                for jc in [
+                    "jeq", "jgt", "jge", "jlt", "jle", "jset", "jne", "jsgt", "jsge", "jslt",
+                    "jsle",
+                ] {
+                    test_ins(
+                        format!(
+                            "{jc} r{src}, r{dst}, l1
+                            or64 r{src},0x12345678
+                            ja l2
+                        l1:
+                            and64 r{dst},0x12345678
+                            ja l2
+                        l2:",
+                        ),
+                        &mut prng,
+                        23,
+                    );
+                }
+            }
+        }
+
+        for _ in 0..10 {
+            let mut imm = prng.next_u32() as i32;
+
+            test_ins(format!("mov64 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("add64 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("sub64 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("xor64 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("and64 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("lmul64 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("uhmul64 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("shmul64 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("udiv64 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("urem64 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("sdiv64 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("srem64 r{src}, {imm}"), &mut prng, 21);
+
+            test_ins(format!("mov32 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("add32 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("sub32 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("xor32 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("and32 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("lmul32 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("udiv32 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("urem32 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("sdiv32 r{src}, {imm}"), &mut prng, 21);
+            test_ins(format!("srem32 r{src}, {imm}"), &mut prng, 21);
```

Comment:
`or32` and `or64` are missing in this section.

`neg32` and `neg64` are missing in the test.